iothub_client/CMakeLists.txt

```diff
@@ -300,20 +300,18 @@ IF(WIN32)
     add_definitions(-DGB_MEASURE_MEMORY_FOR_THIS -DGB_DEBUG_ALLOC)
 ENDIF(WIN32)
 
-if(${build_as_dynamic})
-    add_library(iothub_client SHARED
-        ${iothub_client_c_files} 
-        ${iothub_client_h_files}
-        ./src/iothub_client.def
-    )
-    target_link_libraries(iothub_client ${iothub_client_libs})
+add_library(iothub_client_dll SHARED
```

Comment:
>iothub_client_dll [](start = 12, length = 17)

Do we have to have this all the time.  Should we only include this if you set the option on cmake?